### ISSUE TITLE
404 when trying to view the "Assignment" page for a Company registration application 

### ISSUE THREAD

**Issue by DhanyaHerath** [openmsupply/conforma-web-app#1373]
### Issue 

I saw a `404! Page not found` error when trying to view the "Assignment" page for a Company registration application: 

![image](https://user-images.githubusercontent.com/29744386/193949890-a27bda68-c9af-469f-9d71-bda354931419.png)

I have a suspicion this might only happen for applications without any assignees, since I only got it for certain applications (seemed to break for applications at the Approval stage where there was no outcome yet. 

### Where?
https://fiji.conforma.nz:50000/applications?type=CR1&user-role=reviewer
**Version**: v0.4.5-16
**Snapshot?**: Fiji_2022_10_03_1126

**Comment by CarlosNZ** [comment]
Yeah, good catch. It looks like it's interpreting an empty assignment array as an error. Thanks, will fix.

**Comment by CarlosNZ** [comment]
Hmm, it seems like this application has ended up in a funny state. It's at the "APPROVAL" stage, but it's status is still "DRAFT" which shouldn't be possible. It's not generating any review assignments, which is actually correct for a DRAFT application. Just not sure how it got like that. Did someone manually edit it in the database? @fergie-nz ?

So strictly speaking, the 404 is correct here -- there shouldn't be anything on the Assignment tab as it's still with the applicant. But odd how it got like that in the first place...

**Comment by fergie-nz** [comment]
Certainly possible it was manually changed, but I'm not sure. If it was me, it was inadvertent.

**Comment by CarlosNZ** [comment]
Okay, let's just keep an eye on it and see if it happens again. Could just be an old application that got messed around during updates, etc.

**Comment by DhanyaHerath** [comment]
Not sure if it is helpful, but all of the company applications which don't have an OUTCOME give the same error:

![image](https://user-images.githubusercontent.com/29744386/193979815-55a4eddf-8e8c-43ea-890b-d8fed95d73de.png)

**Comment by jagoje** [comment]
I think these applications were broken due to being uploaded being between builds.. I don't believe this is an issue any longer?

**Comment by nmadruga** [comment]
> I have a suspicion this might only happen for applications without any assignees

@DhanyaHerath @jagoje 
Yes, we do that when there isn't a possible reviewer (no review_assignments) the NoMatch page is displayed with the 404 error. Maybe we can improve this page to show something more accurate as "No assignments available"?

**Comment by DhanyaHerath** [comment]
@nmadruga Yeah absolutely! Something that is clear to users like "There are no reviewers available to assign. Please check [something something action they can do to fix it]
